lightningd/log.c

```diff
@@ -614,7 +614,7 @@ static char *arg_log_prefix(const char *arg, struct log *log)
 
 static void show_log_prefix(char buf[OPT_SHOW_LEN], const struct log *log)
 {
-	strncpy(buf, log->prefix->prefix, OPT_SHOW_LEN);
+	strncpy(buf, log->prefix->prefix, OPT_SHOW_LEN-1);
```

Comment:
Yes, the usage description in ccan/opt/opt.h says:

```
Similarly, if @show is not NULL, it should be of type "void *show(char *, const type *)".  It should write up to OPT_SHOW_LEN bytes into the first argument; unless it uses the entire OPT_SHOW_LEN bytes it should nul-terminate that buffer.
```

So, this should also set `buf[OPT_SHOW_LEN-1] = '\0';`.

Reply:
https://github.com/ElementsProject/lightning/blob/bf80565bc5f958be1d562b15ebd18586d8b0820e/lightningd/log.c#L596-L605

Thanks for the response, same then for `show_log_level` I guess.